### PR TITLE
NDRS-648: Validate Get-Block Response

### PR DESCRIPTION
**Pull request by darthsiroftardis**
REF: https://casperlabs.atlassian.net/browse/NDRS-648

### CODE REVIEW

File: node/src/types/block.rs [hunk 1]
```diff
@@ -1026,6 +1026,20 @@ pub(crate) mod json_compatibility {
         }
     }
 
+    impl From<JsonEraEnd> for EraEnd {
+        fn from(era_end: JsonEraEnd) -> Self {
+            let equivocators = era_end.equivocators;
+            let mut rewards: BTreeMap<PublicKey, u64> = BTreeMap::new();
+            for reward in era_end.rewards {
+                rewards.insert(reward.validator, reward.amount);
+            }
```
Comment (Fraser999) [inline node/src/types/block.rs:1035]: Maybe slightly more canonical to map between two different collections like this:
```suggestion
            let rewards = era_end
                .rewards
                .into_iter()
                .map(|reward| (reward.validator, reward.amount))
                .collect();
```
No need for a `mut` variable then.

**Review by Fraser999 — APPROVED** [review]
One minor nitpick - otherwise LGTM.

File: node/src/types/block.rs (hunk 1 above)
Comment (darthsiroftardis) [reply node/src/types/block.rs]: Applied suggestion

**Pull request merged**